packages/orama/src/trees/avl.ts

```diff
@@ -270,21 +269,19 @@ export function insert<K, V>(root: Node<K, V>, key: K, value: V): Node<K, V> {
     const balanceFactor = getBalanceFactor(parent);
     
     if (balanceFactor === BALANCE_STATE.UNBALANCED_LEFT) {
-        if (key < (parent.left as Node<K, V>).key) {
-            parent = rotateRight(parent);
-        } else {
+        if (key >= (parent.left as Node<K, V>).key) {
             parent.left = rotateLeft(parent.left as Node<K, V>);
-            parent = rotateRight(parent);
         }
+
+        parent = rotateRight(parent);
     }
 
     if (balanceFactor === BALANCE_STATE.UNBALANCED_RIGHT) {
-        if (key > (parent.right as Node<K, V>).key) {
-            parent = rotateLeft(parent);
-        } else {
+        if (key <= (parent.right as Node<K, V>).key) {
             parent.right = rotateRight(parent.right as Node<K, V>);
-            parent = rotateLeft(parent);
         }
+
+        parent = rotateLeft(parent);
     }
 
     if (parent == root) {
```

Comment:
@micheleriva if we want to check the two object are the same reference we can use `===`.
If we want to check if they have equal value it's a bit more hacky